x-pack/plugins/maps/public/kibana_services.ts

```diff
@@ -63,23 +59,27 @@ export const getMapAppConfig = () => mapAppConfig;
 export const getShowMapsInspectorAdapter = () => getMapAppConfig().showMapsInspectorAdapter;
 export const getPreserveDrawingBuffer = () => getMapAppConfig().preserveDrawingBuffer;
 
-// map.* kibana.yml settings from maps_ems plugin that are shared between OSS map visualizations and maps app
-let kibanaCommonConfig: MapsEmsConfig;
-export const setKibanaCommonConfig = (config: MapsEmsConfig) => (kibanaCommonConfig = config);
-export const getKibanaCommonConfig = () => kibanaCommonConfig;
-
+let mapsEms: MapsEmsPluginPublicStart;
 let emsSettings: EMSSettings;
-export const setEMSSettings = (value: EMSSettings) => {
-  emsSettings = value;
+export const setMapsEmsStart = (value: MapsEmsPluginPublicStart) => {
+  mapsEms = value;
+  emsSettings = mapsEms.createEMSSettings();
 };
-export const getEMSSettings = () => {
+// map.* kibana.yml settings from maps_ems plugin that are shared between OSS map visualizations and maps app
+export const getMapsEmsStart: () => MapsEmsPluginPublicStart = () => {
+  return mapsEms;
+};
+
+export const getEMSSettings: () => EMSSettings = () => {
   return emsSettings;
 };
 
-export const getEmsTileLayerId = () => getKibanaCommonConfig().emsTileLayerId;
+export const getMapConfig = () => mapsEms.config;
```

Comment:
getMapConfig method does not need to be exported, its not used outside of this file.